akka-typed/src/main/scala/akka/typed/EventStream.scala

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.typed
+
+import akka.{ event ⇒ e }
+import akka.event.Logging.{ LogEvent, LogLevel, StdOutLogger }
+import akka.testkit.{ EventFilter, TestEvent ⇒ TE }
+import scala.annotation.tailrec
+
+/**
+ * An EventStream allows local actors to register for certain message types, including
+ * their subtypes automatically. Publishing events will broadcast them to all
+ * currently subscribed actors with matching subscriptions for the event type.
+ *
+ * IMPORTANT NOTICE
+ *
+ * This EventStream is local to the ActorSystem, it does not span a cluster. For
+ * disseminating messages across a cluster please refer to the DistributedPubSub
+ * module.
+ */
+trait EventStream {
```

Comment:
Technically it could—perhaps this should be a future refactoring? The only incompatible change would be that querying the logLevel would become asynchronous.